### ISSUE TITLE
Chapter 6 DCGAN.ipynb not working as expected any more

### ISSUE THREAD

**Issue by luke1961** [PacktPublishing/Deep-Learning-with-TensorFlow-2-and-Keras#7]
I use Python 3.8 and Tensorflow 2.4 and I see that at about 1000 epochs discriminator's accuracy reaches 100% and the generator's loss significantly increases and the quality of the images produced by this DCGAN goes down and never recovers. I have tried multiple times, both on my work computer and with the use of colab. In other words the training goes very differently than in [https://github.com/PacktPublishing/Deep-Learning-with-TensorFlow-2-and-Keras/blob/master/Chapter%206/DCGAN.ipynb](https://github.com/PacktPublishing/Deep-Learning-with-TensorFlow-2-and-Keras/blob/master/Chapter%206/DCGAN.ipynb). My presumption is that some improvements in TensorFlow 2.4 cause the generator to train too well and/or too quickly.  Any ideas how to fix DCGAN.ipynb?

**Comment by Fardeenkhan1** [comment]
Do you find the solution why this is happening I am also stuck here.

**Comment by gmcmacran** [comment]
For what it is worth, this is happening to me in 2023 with tensorflow 2.10. Result quality changes run to run. Some runs produce images that resemble a few numbers, but completely missing other numbers. Some runs produce images that don't resemble the numbers at all. Just random black and white pictures.
